src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt

```diff
@@ -144,8 +144,13 @@ class EcommerceTransactionDataProvider(
                         },
                         ::Pair
                     )
-                    .map { (baseTransaction, email) ->
-                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email)
+                    .zipWith(events.collectList(), ::Pair)
+                    .map { (baseTransactionAndEmailPair, events) ->
+                        baseTransactionToTransactionInfoDtoV1(
+                            baseTransactionAndEmailPair.first,
+                            baseTransactionAndEmailPair.second,
+                            events
+                        )
```

Comment:
what about mapping the above Pair to a Triple and use arguments deconstructing for making code clearer?
Something like:
```suggestion
                     .flatMap { (baseTransaction, email) ->
                        events.collectList().map { Triple(baseTransaction, email, it) }
                    }
                    .map { (baseTransaction, email, events) ->
                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email, events)
                    }
```

in this way we can avoid calling `first` and `second` method on the pair class and use the Triple argument deconstructing making code easier to refactoring and removing the variable name baseTransactionAndEmailPair / first and second methods link